src/goto-programs/property_checker.h

```diff
@@ -48,6 +38,28 @@ class property_checkert:public messaget
   typedef std::map<irep_idt, property_statust> property_mapt;
   property_mapt property_map;
 
+  /// return the overall result; this is
+  /// ERROR if any of the results is ERROR, otherwise
+  /// FAIL if any of the results is FAIL, otherwise
+  /// UNKNOWN if any of the results is UNKNOWN, otherwise
+  /// PASS.
+  resultt operatort() const;
+
+  void initialize(const goto_functionst &);
+};
+
+class property_checkert : public messaget
```

Comment:
So what is this class actually good for? The use of `operator()` instead of a named method unfortunately does not add clarity either.